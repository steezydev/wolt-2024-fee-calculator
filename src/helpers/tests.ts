const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
