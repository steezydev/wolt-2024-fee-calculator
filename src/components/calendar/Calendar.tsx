import { setDateDate } from '@/helpers/date';
import { CalendarProps } from '@/types/props/CalendarProps';
import { enGB } from 'date-fns/locale';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

const Calendar = ({ selectedDate, onChange }: CalendarProps) => {
  const handleDateChange = (date: Date | null) => {
    if (date == null) return;
    onChange?.(setDateDate(selectedDate ?? new Date(), date));
  };

  return (
    <DatePicker
      selected={selectedDate}
      onChange={handleDateChange}
      locale={enGB}
      inline
    />
  );
};

export default Calendar;
