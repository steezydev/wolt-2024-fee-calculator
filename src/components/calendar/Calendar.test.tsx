import Calendar from './Calendar';
import { calendarEventTests, calendarRenderTests } from './Calendar.tests';

// Mocking current date to 01.01.2024
beforeEach(() => {
  jest.useFakeTimers();
  jest.setSystemTime(new Date('2024-01-01'));
});

afterEach(() => {
  jest.useRealTimers();
});

calendarRenderTests('Calendar', Calendar);
calendarEventTests('Calendar', Calendar);
