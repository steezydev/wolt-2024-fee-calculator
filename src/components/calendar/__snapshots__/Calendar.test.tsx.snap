// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar when asked to render renders 1`] = `
<div
  style={
    {
      "display": "contents",
    }
  }
>
  <div
    className="react-datepicker"
  >
    <span
      aria-live="polite"
      className="react-datepicker__aria-live"
      role="alert"
    />
    <button
      aria-label="Previous Month"
      className="react-datepicker__navigation react-datepicker__navigation--previous"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="react-datepicker__navigation-icon react-datepicker__navigation-icon--previous"
      >
        Previous Month
      </span>
    </button>
    <button
      aria-label="Next Month"
      className="react-datepicker__navigation react-datepicker__navigation--next"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="react-datepicker__navigation-icon react-datepicker__navigation-icon--next"
      >
        Next Month
      </span>
    </button>
    <div
      className="react-datepicker__month-container"
    >
      <div
        className="react-datepicker__header "
      >
        <div
          className="react-datepicker__current-month"
        >
          January 2024
        </div>
        <div
          className="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
          onFocus={[Function]}
        />
        <div
          className="react-datepicker__day-names"
        >
          <div
            className="react-datepicker__day-name"
          >
            Mo
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Tu
          </div>
          <div
            className="react-datepicker__day-name"
          >
            We
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Th
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Fr
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Sa
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Su
          </div>
        </div>
      </div>
      <div
        aria-label="month  2024-01"
        className="react-datepicker__month"
        onMouseLeave={[Function]}
        role="listbox"
      >
        <div
          className="react-datepicker__week react-datepicker__week--keyboard-selected"
        >
          <div
            aria-current="date"
            aria-disabled={false}
            aria-label="Choose Monday, 1 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--keyboard-selected react-datepicker__day--today"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={0}
            title=""
          >
            1
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 2 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--002"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            2
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 3 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--003"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            3
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 4 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--004"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            4
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 5 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--005"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            5
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 6 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--006 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            6
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 7 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--007 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            7
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 8 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--008"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            8
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 9 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--009"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            9
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 10 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--010"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            10
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 11 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--011"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            11
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 12 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--012"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            12
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 13 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--013 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            13
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 14 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--014 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            14
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 15 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--015"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            15
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 16 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--016"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            16
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 17 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--017"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            17
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 18 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--018"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            18
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 19 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--019"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            19
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 20 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--020 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            20
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 21 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--021 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            21
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 22 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--022"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            22
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 23 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--023"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            23
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 24 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--024"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            24
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 25 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--025"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            25
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 26 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--026"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            26
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 27 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--027 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            27
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 28 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--028 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            28
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 29 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--029"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            29
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 30 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--030"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            30
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 31 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--031"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            31
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 1 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            1
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 2 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            2
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 3 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--003 react-datepicker__day--weekend react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            3
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 4 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--004 react-datepicker__day--weekend react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Calendar when asked to render with selected date renders with selected date 1`] = `
<div
  style={
    {
      "display": "contents",
    }
  }
>
  <div
    className="react-datepicker"
  >
    <span
      aria-live="polite"
      className="react-datepicker__aria-live"
      role="alert"
    />
    <button
      aria-label="Previous Month"
      className="react-datepicker__navigation react-datepicker__navigation--previous"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="react-datepicker__navigation-icon react-datepicker__navigation-icon--previous"
      >
        Previous Month
      </span>
    </button>
    <button
      aria-label="Next Month"
      className="react-datepicker__navigation react-datepicker__navigation--next"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="react-datepicker__navigation-icon react-datepicker__navigation-icon--next"
      >
        Next Month
      </span>
    </button>
    <div
      className="react-datepicker__month-container"
    >
      <div
        className="react-datepicker__header "
      >
        <div
          className="react-datepicker__current-month"
        >
          January 2024
        </div>
        <div
          className="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
          onFocus={[Function]}
        />
        <div
          className="react-datepicker__day-names"
        >
          <div
            className="react-datepicker__day-name"
          >
            Mo
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Tu
          </div>
          <div
            className="react-datepicker__day-name"
          >
            We
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Th
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Fr
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Sa
          </div>
          <div
            className="react-datepicker__day-name"
          >
            Su
          </div>
        </div>
      </div>
      <div
        aria-label="month  2024-01"
        className="react-datepicker__month"
        onMouseLeave={[Function]}
        role="listbox"
      >
        <div
          className="react-datepicker__week"
        >
          <div
            aria-current="date"
            aria-disabled={false}
            aria-label="Choose Monday, 1 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--today"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            1
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 2 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--002"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            2
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 3 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--003"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            3
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 4 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--004"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            4
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 5 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--005"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            5
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 6 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--006 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            6
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 7 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--007 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            7
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 8 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--008"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            8
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 9 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--009"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            9
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 10 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--010"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            10
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 11 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--011"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            11
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 12 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--012"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            12
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 13 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--013 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            13
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 14 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--014 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            14
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 15 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--015"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            15
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 16 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--016"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            16
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 17 January 2024"
            aria-selected={true}
            className="react-datepicker__day react-datepicker__day--017 react-datepicker__day--selected"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={0}
            title=""
          >
            17
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 18 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--018"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            18
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 19 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--019"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            19
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 20 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--020 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            20
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 21 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--021 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            21
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 22 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--022"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            22
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 23 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--023"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            23
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 24 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--024"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            24
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 25 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--025"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            25
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 26 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--026"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            26
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 27 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--027 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            27
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 28 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--028 react-datepicker__day--weekend"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            28
          </div>
        </div>
        <div
          className="react-datepicker__week"
        >
          <div
            aria-disabled={false}
            aria-label="Choose Monday, 29 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--029"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            29
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Tuesday, 30 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--030"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            30
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Wednesday, 31 January 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--031"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            31
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Thursday, 1 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            1
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Friday, 2 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            2
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Saturday, 3 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--003 react-datepicker__day--weekend react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            3
          </div>
          <div
            aria-disabled={false}
            aria-label="Choose Sunday, 4 February 2024"
            aria-selected={false}
            className="react-datepicker__day react-datepicker__day--004 react-datepicker__day--weekend react-datepicker__day--outside-month"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            role="option"
            tabIndex={-1}
            title=""
          >
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
